Shape_detection/include/CGAL/Shape_detection/Region_growing/Triangle_mesh/Least_squares_plane_fit_region.h

```diff
@@ -0,0 +1,458 @@
+// Copyright (c) 2018 INRIA Sophia-Antipolis (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Florent Lafarge, Simon Giraudot, Thien Hoang, Dmitry Anisimov
+//
+
+#ifndef CGAL_SHAPE_DETECTION_REGION_GROWING_TRIANGLE_MESH_LEAST_SQUARES_PLANE_FIT_REGION_H
+#define CGAL_SHAPE_DETECTION_REGION_GROWING_TRIANGLE_MESH_LEAST_SQUARES_PLANE_FIT_REGION_H
+
+#include <CGAL/license/Shape_detection.h>
+
+// Internal includes.
+#include <CGAL/Shape_detection/Region_growing/internal/property_map.h>
+
+namespace CGAL {
+namespace Shape_detection {
+namespace Triangle_mesh {
```

Comment:
I would disagree with renaming it to `Triangle_mesh`. In fact, all classes here work with a general face graph and I do not see why it cannot be used on a surface mesh with non-triangle faces? The function `create_plane_from_faces()` in `utils.h` that fits a plane first collects all points of the face and then applies a plane fitting so it does not matter how many vertices a face has. Even the new class `Polyline_graph` works with a face graph. But this new name will lead users to the conclusion that they need to search somewhere else to solve their problem in case they have a non-triangle mesh. I think I even had an example/test at some point that applied RG to a non-triangulated cube.

Reply:
I agree. We were thinking to use the linear_least_squares_fitting_3 with triangles instead of the vertices. This uses a weight to take into account the different size of triangles. However, we finally kept your way and so I just reverted back to Polygon_mesh.